### PR TITLE
Create object storage file formats page

### PR DESCRIPTION
**Pull request by m57lyra**
## Description

This PR pulls information from connector pages and centralizes it.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: docs/src/main/sphinx/connector/delta-lake.rst [hunk 1]
```diff
@@ -25,6 +25,8 @@ To connect to Databricks Delta Lake, you need:
 * Access to the Hive metastore service (HMS) of Delta Lake or a separate HMS.
 * Network access to the HMS from the coordinator and workers. Port 9083 is the
   default port for the Thrift protocol used by the HMS.
+* Data files must be in a
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/delta-lake.rst:28]: Spell out what they are and link to the specific sections only

File: docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -17,6 +17,7 @@ Hive connector
     IBM Cloud Object Storage <hive-cos>
     Storage Caching <hive-caching>
     Alluxio <hive-alluxio>
+    Object storage file formats <object-storage-file-formats>
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:20]: I guess this is fine for now .. but needs to move outside of the hive context over time

File: docs/src/main/sphinx/connector/hive.rst [hunk 3]
```diff
@@ -54,6 +55,20 @@ The coordinator and all workers must have network access to the Hive metastore
 and the storage system. Hive metastore access with the Thrift protocol defaults
 to using port 9083.
 
+Data files must be in a
+:doc:`supported file format <./object-storage-file-formats>`. Supported file
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:59]: remove this link

File: docs/src/main/sphinx/connector/hive.rst [hunk 4]
```diff
@@ -54,6 +55,20 @@ The coordinator and all workers must have network access to the Hive metastore
...
+formats include:
+
+* ORC
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hive.rst:62]: Make each one of these here link

File: docs/src/main/sphinx/connector/hudi.rst [hunk 5]
```diff
@@ -17,6 +17,8 @@ To use the Hudi connector, you need:
 * Network access from the Trino coordinator and workers to the Hudi storage.
 * Access to the Hive metastore service (HMS).
 * Network access from the Trino coordinator to the HMS.
+* Data files must be in the
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/hudi.rst:20]: Spell out which specific ones are supported and link directly (ORC and Parquet only)

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 6]
```diff
@@ -42,8 +42,12 @@ To use Iceberg, you need:
   :ref:`AWS Glue catalog<iceberg-glue-catalog>`, a :ref:`JDBC catalog
   <iceberg-jdbc-catalog>`, a :ref:`REST catalog<iceberg-rest-catalog>`, or a
   :ref:`Nessie server<iceberg-nessie-catalog>`.
-* Network access from the Trino coordinator to the HMS. Hive metastore access
-  with the Thrift protocol defaults to using port 9083.
+* Data files must be in a
+  :doc:`supported file format <./object-storage-file-formats>`. These include:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:46]: Remove link

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 7]
```diff
@@ -42,8 +42,12 @@ To use Iceberg, you need:
...
+  
+  - ORC
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:48]: Link all these

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 8]
```diff
@@ -42,8 +42,12 @@ To use Iceberg, you need:
...
+  - Parquet
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:49]: Add that this is the default

File: docs/src/main/sphinx/connector/iceberg.rst [hunk 9]
```diff
@@ -42,8 +42,12 @@ To use Iceberg, you need:
...
+  - Avro
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/iceberg.rst:50]: Not sure but I think Avro is only for metadata and thats an implementation detail and should be left out here

File: docs/src/main/sphinx/connector/object-storage-file-formats.rst [hunk 10]
```diff
@@ -0,0 +1,112 @@
+===========================
+Object storage file formats
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/object-storage-file-formats.rst:2]: Add a section for each format that at least says what it is and has an anchor for the header .. and then potentially more config stuff

File: docs/src/main/sphinx/release/release-332.md [hunk 11]
```diff
@@ -60,7 +60,7 @@
   `hive.metastore.glue.endpoint-url` configuration property. ({issue}`3239`)
 - Add experimental file system caching. This can be enabled with the `hive.cache.enabled` configuration property. ({issue}`2679`)
 - Support reading files compressed with newer versions of LZO. ({issue}`3209`)
-- Add support for {ref}`alluxio-catalog-service`. ({issue}`2116`)
+- Add support for Alluxio Catalog Service. ({issue}`2116`)
```
Comment (mosabua) [inline docs/src/main/sphinx/release/release-332.md:63]: Unrelated .. probably need to rebase

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/hive.rst (hunk 2 above)
Comment (m57lyra) [reply docs/src/main/sphinx/connector/hive.rst:20]: Yes. Phase 4 is re-org.

File: docs/src/main/sphinx/release/release-332.md (hunk 11 above)
Comment (m57lyra) [reply docs/src/main/sphinx/release/release-332.md:63]: Still fails after rebase: Warning, treated as error:
/docs/src/main/sphinx/release/release-332.md:63:undefined label: alluxio-catalog-service

File: docs/src/main/sphinx/connector/object-storage-file-formats.rst [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+        equivalent catalog session property is ``orc_use_column_names``.
+      - ``false``
+    * - ``hive.orc.bloom-filters.enabled``
+      - Enable bloom filters for predicate pushdown.
+      - ``false``
+    * - ``hive.orc.read-legacy-short-zone-id``
+      - Allow reads on ORC files with short zone ID in the stripe footer.
+      - ``false``
+
+.. _hive-parquet-configuration:
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/object-storage-file-formats.rst:47]: just parquet

File: docs/src/main/sphinx/connector/object-storage-file-formats.rst [hunk 13]
```diff
@@ -0,0 +1,94 @@
...
+Object storage file formats
+===========================
+
+Object storage connectors support one or more file formats specified by the
+underlying data source.
+
+In the case of serializable formats, only specific
+`SerDes <https://www.wikipedia.org/wiki/SerDes>`_ are allowed:
+
+* RCText - RCFile ``ColumnarSerDe``
```
Comment (mosabua) [inline docs/src/main/sphinx/connector/object-storage-file-formats.rst:11]: These might be wrong now with the refactor .. but I think we should merge like this and then improve more after

**Review by mosabua — CHANGES_REQUESTED** [review]

File: docs/src/main/sphinx/connector/delta-lake.rst (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/connector/delta-lake.rst:28]: Seems like its only Parquet - https://docs.delta.io/latest/delta-faq.html

**Comment by m57lyra** [comment]
https://github.com/trinodb/trino/pull/17510#discussion_r1294905622

^^ That is something to be worked out in another PR. The intent of these PRs were to work with what we have, not to make changes to technical information @mosabua

**Review by mosabua — COMMENTED** [review]
Looks good now. Further cleanup will follow in later PRs.

**Comment by ebyhr** [comment]
@mosabua Please remove a PR number from the commit title when you use "Squash and merge"
Screenshot 2023-08-17 at 8 22 27

**Comment by mosabua** [comment]
Dang .. I thought I did .. so weird

**Pull request merged**